### PR TITLE
[CCBC-1447] Fixes required to build examples with CMake

### PR DESCRIPTION
**Pull request by raycardillo**
This is a substantial improvement but there is still more work to do. However, this should unblock anyone who wants to make further contributions to help improve the examples in the near future.

Some quick notes:
1. I intended to review all examples, but it was ultimately more than I could do in one pass, so I commented out the build for any examples that need additional refactoring.
2. The most significant contribution is the improvement to the `CMake` configuration, but there is more work needed for the other projects in the examples sub-dirs also.
3. These examples are used with the docs-site build system as well, so we should definitely spot check the docs-site build as well. It built fine locally for me, and I looked at a few quickly, but it can use a second pair of eyes.

### CODE REVIEW

**Comment by osfameron** [comment]
Adding Maria, Richard ~and Amarantha~ for review  ~as you mention need to double-check `docs-site` system!~

Thanks Ray, checking this out now.

File: modules/devguide/examples/c/bincoding/README.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 This example demonstrates how to write custom transcoders for structs in C.
 
-Here we store user profile with embedded avatar as document in Couchbase, and retrieve it back to render as HTML.
-Note, that encoding of the `struct Profile` remains friendly to Couchbase indexers, and data could be retrieven using any of queries (N1QL, FTS etc).
+Here we store user profile with embedded avatar as document in Couchbase, and retrieve it back to render as HTML. Note,
+that encoding of the `struct Profile` remains friendly to Couchbase indexers, and data could be retrieven using any of
+queries (N1QL, FTS etc).
 
 To try the example, use Makefile to build the binary and then execute it:
```
Comment (osfameron) [inline modules/devguide/examples/c/bincoding/README.md:7]: reference to "Makefile" now needs updating

**Review by osfameron — APPROVED** [review]
This looks great, thanks @raycardillo! Can confirm it now builds-on-my-machine (also OSX).

A lot of the diff is formatting changes, which means it's hard(er) for me to understand what changes were made to fix the build. That's fine, but:

* if there's any specific changes you think we should be aware of, please do mention
* are style changes discussed with @avsej, and to be used going forward (and all existing code samples conformed to it)
* does this style have a name/Editorconfig setting so it can be easily replicated?
* general comment: I like how there is a little more spacing in this style. (Though, oddly, not for `case` statements?)

**Review by RichardSmedley — APPROVED** [review]
LGTM - thanks @raycardillo 

Feel free to merge @osfameron if you're satisfied - let's get these fixes onto GH  :+1:

**Comment by osfameron** [comment]
Discussing with Richard, I realise I've misparsed the reference to "docs-site" above..., sorry for ping @amarantha-k, removing you from reviewers.

File: modules/devguide/examples/c/bincoding/README.md (hunk 1 above)
Comment (raycardillo) [reply modules/devguide/examples/c/bincoding/README.md:7]: So, the subdirs still need more work. These still will not build. So I didn't touch them at all. Maybe we break that out into another ticket to track the additional cleanup required? I just wasn't sure if we might already have some  (I might have already resolved some other issues in this pass).

**Comment by raycardillo** [comment]
> This looks great, thanks @raycardillo! Can confirm it now builds-on-my-machine (also OSX).
> 
> A lot of the diff is formatting changes, which means it's hard(er) for me to understand what changes were made to fix the build. That's fine, but:
> 
> * if there's any specific changes you think we should be aware of, please do mention
> * are style changes discussed with @avsej, and to be used going forward (and all existing code samples conformed to it)
> * does this style have a name/Editorconfig setting so it can be easily replicated?
> * general comment: I like how there is a little more spacing in this style. (Though, oddly, not for `case` statements?)

Sorry for the style changes, I noticed there were a lot of inconsistencies, so I tried to configure the style formatter to be consistent with what I saw in the main LCB source code base. However, I might have missed some special cases. I will discuss with @avsej and we can export a CLion style format so they're all consistent in the future.

For a few cases I fixed, nothing special to note, in many cases, just porting the example code to use the latest version of the API. So I had to refactor the instance variables, some structs that are now internal, some method names, etc. However, I didn't refactor any logic other than that.

File: modules/devguide/examples/c/bincoding/README.md (hunk 1 above)
Comment (osfameron) [reply modules/devguide/examples/c/bincoding/README.md:7]: ah, gotcha. I added a commit in https://github.com/couchbase/docs-sdk-c/pull/112 but sounds like I jumped the gun, so I'll rebase it out now...

**Pull request merged**